### PR TITLE
Make the Dialog mobile friendly

### PR DESCRIPTION
**Pull request by Rowno**
This change makes the Dialog resize gracefully to fit the available viewport.

It should be a non-breaking change and the Dialog should behave the same on desktop as it did before.

kapture 2018-08-23 at 0 21 39

### CODE REVIEW

**Comment by mshwery** [comment]
@Rowno I think once we get to a reasonably small viewport width (or height) we should reduce the margin/padding between the edge of the screen and the dialog.

File: src/dialog/src/Dialog.js [hunk 1]
```diff
@@ -209,94 +217,91 @@ class Dialog extends React.Component {
...
-    if (Number.isInteger(topOffset)) {
-      maxHeight = `calc(100vh - ${topOffset}px)`
-    } else {
-      maxHeight = `calc(100vh - ${topOffset})`
-    }
+    const topOffsetWithUnit = Number.isInteger(topOffset)
+      ? `${topOffset}px`
+      : topOffset
+    const sideOffsetWithUnit = Number.isInteger(sideOffset)
+      ? `${sideOffset}px`
```
Comment (jeroenransijn) [inline src/dialog/src/Dialog.js]: I don't think we need a unit here

File: src/dialog/src/Dialog.js [hunk 2]
```diff
@@ -209,94 +217,91 @@ class Dialog extends React.Component {
...
     } = this.props
 
-    let maxHeight
-    if (Number.isInteger(topOffset)) {
-      maxHeight = `calc(100vh - ${topOffset}px)`
-    } else {
-      maxHeight = `calc(100vh - ${topOffset})`
-    }
+    const topOffsetWithUnit = Number.isInteger(topOffset)
+      ? `${topOffset}px`
```
Comment (jeroenransijn) [inline src/dialog/src/Dialog.js]: I don't think we need a unit here

Comment (Rowno) [reply src/dialog/src/Dialog.js]: We do need this one for the `maxHeight` `calc()` below.

Comment (jeroenransijn) [reply src/dialog/src/Dialog.js]: `${topOffset}px` => `topOffset` right?

Comment (Rowno) [reply src/dialog/src/Dialog.js]: No, because then `calc(100% - ${topOffsetWithUnit} * 2)` won't work.

**Pull request merged**